enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/AuthScenariosLogic.java

```diff
@@ -386,14 +390,24 @@ public void roleManagement6() throws Throwable
         S henrik = neo.login( "Henrik", "bar" );
         neo.assertAuthenticated( henrik );
 
+        final long t0 = System.currentTimeMillis();
+        final long INITIAL_NUM_NODES = 3;
+        final long LOAD_TIMEOUT_MILLIS  = 2000;
+        final int ROWS_IN_CSV = 10;
+
         ThreadedTransactionPeriodicCommit<S> perCommit = new ThreadedTransactionPeriodicCommit<>( neo );
-        perCommit.execute( threading, henrik );
+        perCommit.execute( threading, henrik, ROWS_IN_CSV );
         perCommit.barrier.await();
 
-        // Would prefer something more robust here, but could not get it to work
-        // This test might turn flaky if committing the initial lines takes longer than 100 ms
-        // Ideally we would be polling for the correct number of nodes, given some timeout
-        Thread.sleep( 200 );
+        long nodeCount = INITIAL_NUM_NODES;
+        while(nodeCount <= INITIAL_NUM_NODES) {
+            if (System.currentTimeMillis() - t0 > LOAD_TIMEOUT_MILLIS )
+            {
+                fail("No nodes added from LOAD CSV within " + LOAD_TIMEOUT_MILLIS / 1000 +" seconds");
```

Comment:
Even though this is much less likely to fail than before, it still looks flaky by design.